prov/tcp/src/tcpx.h

```diff
@@ -117,8 +149,15 @@ struct poll_fd_mgr {
 	int			max_nfds;
 };
 
+struct tcpx_conn_req {
+	uint8_t type;
+	uint8_t rsvd;
+	uint16_t cm_data_sz;
+};
```

Comment:
Since this is a wire protocol message, it needs to be part of the broader protocol run over the provider.  Please look at fi_proto.h for protocol headers used by other providers.  For example, there are enums for ofi_ctrl_connreq, ofi_ctrl_connresp.  We can expand that as needed, though I'm not sure we need a shutdown equivalent.  Also see struct ofi_ctrl_hdr.

One of the problem this structure has is that there is no versioning carried as part of the message.  All protocol messages should include version as the first field.